Cleanup and respec fixes

patrickhlauke opened the pull request:
- add `class="export"` to definitions to export
- add `<dfn>` and `data-lt="..."` synonyms/aliases where appropriate
- eliminate most hardcoded `href="#..."` relative links in favour of autogenerated respec definition links
- remove the "candidate regions" bit from the "Declaring candidate regions for direct manipulation behaviors" heading - now that section is just "Declaring direct manipulation behaviors" (not sure why the "candidate regions" bit was needed, and always sounded a bit too wishy-washy...why "candidates"? also made it easier to then set it as a clean `<dfn>` to link to)
- fix some markup validation issues/misnestings
- fix a broken link to UIEVENTS Mouse Event Order (use `data-cite` not `href` for Respec to do its magic)

Closes https://github.com/w3c/pointerevents/issues/435


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***

### :boom: Error: 502 Bad Gateway :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Jun 1, 2022, 2:56 PM UTC)_.

<details>
<summary>More</summary>


PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fpointerevents%2Ff576688d3c2d267c6ed7f30986eacef1b66ae11f%2Findex.html%3FisPreview%3Dtrue)

```
<html><body><h1>502 Bad Gateway</h1>
The server returned an invalid or incomplete response.
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/pointerevents%23441.)._
</details>

patrickhlauke committed:
Cleanup and respec fixes

- add `class="export"` to definitions to export
- add `<dfn>` and `data-lt="..."` synonims/aliases where appropriate
- eliminate most hardcoded `href="#..."` relative links in favour of autogenerated respec definition links

Closes https://github.com/w3c/pointerevents/issues/435

patrickhlauke commented:
Not a super expert on respec, nor on the whole "which things should/shouldn't be 'exported' from a spec" aspect. so please have a close look particularly at that aspect.

Assuming the changes are good though, this eliminates all previous Respec errors we were having (and tidies up some of the hardcoded relative links, which always irked me a bit)

patrickhlauke commented:
Also shout if there was a reason for having that "candidate regions" part of the heading that escaped me...but to me that was never really necessary?

patrickhlauke committed:
Fix markup validation errors / misnestings

smaug---- commented on index.html:
```diff
@@ -947,13 +945,13 @@ <h3>Implicit release of pointer capture</h3>
     </section>
     <section>
-        <h1><dfn>Coalesced and predicted events</dfn></h1>
+        <h1><dfn class="export">Coalesced and predicted events</dfn></h1>
```
I wonder if this needs to have export. Feels very PointerEvents spec internal thing.

smaug---- commented on index.html:
```diff
@@ -947,13 +945,13 @@ <h3>Implicit release of pointer capture</h3>
 
         <section>
-            <h2><dfn>Coalesced events</dfn></h2>
+            <h2><dfn class="export">Coalesced events</dfn></h2>
```
Same here.

smaug---- commented on index.html:
```diff
@@ -1146,7 +1143,7 @@ <h2><dfn>Predicted events</dfn></h2>
 
         <section>
-            <h2><dfn>Populating and maintaining the coalesced and predicted events lists</dfn></h2>
+            <h2><dfn class="export">Populating and maintaining the coalesced and predicted events lists</dfn></h2>
```
Also this, does this need to have export? Respec documentation says "Only export things that other specifications need!"  (Having export for specific events feels ok)

patrickhlauke commented:
@smaug---- yup, keep 'em coming. i was naively adding exports to "things that somebody may potentially want to point to from the outside", but i probably overestimated how interesting/relevant they might be from an outside perspective.

smaug---- reviewed (approved):
I'd drop perhaps those few exports

patrickhlauke committed:
Remove some unnecessary exports

patrickhlauke committed:
Fix broken link/reference to UIEVENTS Mouse Event Order

this was also preventing pr-preview from running correctly. with thanks to Denis Ah Kang for spotting the broken link.

patrickhlauke commented:
@flackr @mustaqahmed you ok with these changes? i'll merge these end of today unless I hear otherwise if that's ok.

mustaqahmed commented on index.html:
```diff
@@ -771,8 +769,8 @@ <h2>Extensions to the `Navigator` interface</h2>
     <section>
-        <h1>Declaring candidate regions for direct manipulation behaviors</h1>
-        <p>As noted in <a href="#attributes-and-default-actions">Attributes and Default Actions</a>, viewport manipulations (panning and zooming) cannot be suppressed by canceling a pointer event. Instead, authors must explicitly define which of these behaviors they want to allow, and which they want to suppress, using the <code>touch-action</code> CSS property.</p>
+        <h1><dfn data-lt="direct manipulation behavior">Declaring candidate regions for direct manipulation behaviors</dfn></h1>
```
Optional nit: I am a bit inclined towards having the word "declarative" in the anchor label, because that the main behavior diff here vs, say, Touch Events.  What about "declarative direct manipulation" or +" behavior"?

patrickhlauke replied:
how about just "Declaring direct manipulation behavior" ?

patrickhlauke replied:
which i see is what i wanted to change this to originall,y but seem not to have done so in the end...oops

patrickhlauke replied:
made some last tweaks just now to emphasise the "declare/declarative" aspect

mustaqahmed replied:
+1 for "declaring ..." (pun intended :smile:)

mustaqahmed reviewed (approved):
Looks good to me, thanks.

patrickhlauke committed:
Actually remove "candidate regions" from the heading, tweak the anchor and prose to include declare/declaratively

mustaqahmed commented on index.html:
```diff
@@ -394,110 +393,110 @@ <h2>The <dfn>primary pointer</dfn></h2>
                         </table>
 
-                        <p>Viewport manipulations (panning and zooming) — generally, as a result of a <a>direct manipulation</a> interaction — are intentionally NOT a default action of pointer events, meaning that these behaviors (e.g. panning a page as a result of moving a finger on a touchscreen) cannot be suppressed by canceling a pointer event. Authors must instead use <code>touch-action</cod ...
+                        <p>Viewport manipulations (panning and zooming) — generally, as a result of a <a>direct manipulation</a> interaction — are intentionally NOT a default action of pointer events, meaning that these behaviors (e.g. panning a page as a result of moving a finger on a touchscreen) cannot be suppressed by canceling a pointer event. Authors must instead use <code>touch-action</cod ...
```
One more nit, sorry: the "the" in "declare the direct manipulation behavior" sounds a bit odd to me because at this point the behavior is not defined to the reader.  Perhaps this is only me, but calling our regardless.
